pandas/tests/io/formats/test_format.py

```diff
@@ -3224,6 +3234,22 @@ def test_period_tz(self):
             per = dt.to_period(freq="H")
         assert per.format()[0] == "2013-01-01 00:00"
 
+    def test_period_custom_locale(self, overridden_locale):
+        # GH#46319 locale-specific formatting directive leads to non-utf8 str result
```

Comment:
Is this clear enough ? I can otherwise replace `str` with `char*` and replace `formatting` with `strftime`. Let me know

Reply:
I went for such a mod eventually